src/V3AstNodes.cpp

```diff
@@ -1266,50 +1267,99 @@ static std::string nodeAddr(const AstNode* nodep) {
+        if (user4p()) str << " u4=" << nodeAddr(user4p());
+        if (user5p()) str << " u5=" << nodeAddr(user5p());
+        if (hasDType()) {
+            // Final @ so less likely to by accident read it as a nodep
+            if (dtypep() == this) {
+                str << " @dt=this@";
+            } else {
+                str << " @dt=" << nodeAddr(dtypep()) << "@";
+            }
+            if (AstNodeDType* const dtp = dtypep()) dtp->dumpSmall(str);
+        } else {  // V3Broken will throw an error
+            if (dtypep()) str << " %Error-dtype-exp=null,got=" << nodeAddr(dtypep());
         }
-        if (AstNodeDType* const dtp = dtypep()) dtp->dumpSmall(str);
-    } else {  // V3Broken will throw an error
-        if (dtypep()) str << " %Error-dtype-exp=null,got=" << nodeAddr(dtypep());
-    }
-    if (name() != "") {
-        if (VN_IS(this, Const)) {
-            str << "  " << name();  // Already quoted
-        } else {
-            str << "  " << V3OutFormatter::quoteNameControls(name());
+        if (name() != "") {
+            if (VN_IS(this, Const)) {
+                str << "  " << name();  // Already quoted
+            } else {
+                str << "  " << V3OutFormatter::quoteNameControls(name());
+            }
         }
-    }
+    } else {
+        str << "\tn" << nodeAddr(this) << "\t[fontsize=8 "
+              << "label=\"" << typeName() << "\\n" <<name();
+        str << "\"";
+        str << ", color=" << dotColor();
+        str << ", shape=" << dotShape();
+        str << ", style=" << dotStyle();
+        str << "];\n";
+		if (user1p()) {
+			str << "\tn" << nodeAddr(this) << " -> "<< nodeAddr(user1p())
+						  << " ["
+						  << "fontsize=8 label=\"" << "u1" << "\""
+						  << " weight=" << 1 << " color=" << "red";
+			str << "];\n";
+		}
+		if (user2p()) {
+			str << "\tn" << nodeAddr(this) << " -> "<< nodeAddr(user2p())
+						  << " ["
+						  << "fontsize=8 label=\"" << "u2" << "\""
+						  << " weight=" << 1 << " color=" << "red";
+			str << "];\n";
+		}
+        if (user3p()) {
+            str << "\tn" << nodeAddr(this) << " -> "<< nodeAddr(user3p())
+                          << " ["
+                          << "fontsize=8 label=\"" << "u3" << "\""
+                          << " weight=" << 1 << " color=" << "red";
+            str << "];\n";
+        }
+        if (user4p()) {
+            str << "\tn" << nodeAddr(this) << " -> "<< nodeAddr(user4p())
+                          << " ["
+                          << "fontsize=8 label=\"" << "u4" << "\""
+                          << " weight=" << 1 << " color=" << "red";
+            str << "];\n";
+        }
+        if (user5p()) {
+            str << "\tn" << nodeAddr(this) << " -> "<< nodeAddr(user5p())
+                          << " ["
+                          << "fontsize=8 label=\"" << "u5" << "\""
+                          << " weight=" << 1 << " color=" << "red";
+            str << "];\n";
+        }
+	}
```

Comment:
Let's make this part of `AstNode::dumpTreeDot`, so we don't need to add `if (v3Global.opt.dumpTreeDot()) { return; }` everywhere below.